packages/amplify-graphql-api-construct/src/__tests__/amplify-graphql-api.test.ts

```diff
@@ -33,4 +33,73 @@ describe('AmplifyGraphqlApi', () => {
       );
     });
   });
+
+  describe('storeOutput', () => {
+    const tokenRegex = /\$\{Token\[TOKEN\.\d+\]\}/;
+    const awsRegionTokenRegex = /\$\{Token\[AWS\.Region\.\d+\]\}/;
+
+    test('stores output with outputStorageStrategy', () => {
+      const api = new AmplifyGraphqlApi(new cdk.Stack(), 'TestApi', {
+        schema: /* GraphQL */ `
+          type Todo @model @auth(rules: [{ allow: public }]) {
+            description: String!
+          }
+        `,
+        authorizationConfig: {
+          apiKeyConfig: { expires: cdk.Duration.days(7) },
+        },
+      });
+
+      const addBackendOutputEntry = jest.fn();
+      const outputStorageStrategyMock = {
+        addBackendOutputEntry,
+        flush: jest.fn(),
+      };
+      api.storeOutput(outputStorageStrategyMock);
```

Comment:
You can use `Template.from(stack) to render and avoid testing w/ token Regexes 

https://docs.aws.amazon.com/cdk/v2/guide/testing.html#testing_fine_grained

Reply:
Sorry, this points to the long lines, should have pointed to the `tokenRegex` usage.

Reply:
I don't think this will useful here. We are asserting what is passed to the `outputStorageStrategy`. In this case they are refs. 

I'm going to add a test for the default `outputStorageStrategy`, which won't be a mock. In that case we will call `outputStorageStrategy.flush()` and then we will be able to render the template and see the stored metadata.